### ISSUE TITLE
`explore`: add configurable key bindings (and default `vim` bindings?)

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#7819]
### Related problem

i'm VERY much used to the `vim` motions, i.e. the keys that need to be pressed to move around in a text buffer, e.g. `h`, `j`, `k` and `l` to go left, down, up and right, `gg` and `G` to go to the top and the bottom, `w`  and `b` to go one word forward and backward, `0` and `$` to go to the beginning of a line or to its end, etc, etc...

for me, and i think all the other people used to `vim`-like motions, `explore` can be a bit confusig...
i need to use the arrows to move around, not the most convenient :yum: 

> **Note**  
> if you're interested in `vim` motions
> please feel free to ask any additional information

> **Note**  
> if i'm not clear enough about the "configurable" part of my request below, please feel free to tell me :yum:  

### Describe the solution you'd like

i see two points about this

- **make `vim` motion the default bindings**: this would be perfect for me and all the *vimmers* around here BUT i do not want to impose that, that might be too much, i just put that here so we can discuss this :relieved: 
- **make the motions configurable**: given the output of `:config-show` from inside `explore`
```
╭───────────────────────┬───────────────────────────────────────╮
│                       │ ╭────┬────────╮                       │
│ highlight             │ │ bg │ yellow │                       │
│                       │ │ fg │ black  │                       │
│                       │ ╰────┴────────╯                       │
│                       │ ╭──────────────────┬─────────╮        │
│ table                 │ │ split_line       │ #404040 │        │
│                       │ │ line_head_top    │ true    │        │
│                       │ │ line_head_bottom │ true    │        │
│                       │ │ show_index       │ true    │        │
│                       │ │ cursor           │ true    │        │
│                       │ │ line_index       │ true    │        │
│                       │ │ show_head        │ true    │        │
│                       │ │ line_shift       │ true    │        │
│                       │ ╰──────────────────┴─────────╯        │
│ help_banner           │ true                                  │
│                       │ ╭────┬─────────╮                      │
│ status_bar_background │ │ fg │ #1D1F21 │                      │
│                       │ │ bg │ #C4C9C6 │                      │
│                       │ ╰────┴─────────╯                      │
│                       │ ╭──────────────┬────────────────────╮ │
│ try                   │ │              │ ╭──────┬─────────╮ │ │
│                       │ │ border_color │ │ bg   │         │ │ │
│                       │ │              │ │ fg   │ #C0CAF5 │ │ │
│                       │ │              │ │ attr │         │ │ │
│                       │ │              │ ╰──────┴─────────╯ │ │
│                       │ ╰──────────────┴────────────────────╯ │
│                       │ ╭──────────────┬────────────────────╮ │
│ config                │ │              │ ╭──────┬─────────╮ │ │
│                       │ │ border_color │ │ bg   │         │ │ │
│                       │ │              │ │ fg   │ #C0CAF5 │ │ │
│                       │ │              │ │ attr │         │ │ │
│                       │ │              │ ╰──────┴─────────╯ │ │
│                       │ │              │ ╭────┬────────╮    │ │
│                       │ │ cursor_color │ │ bg │ yellow │    │ │
│                       │ │              │ │ fg │ black  │    │ │
│                       │ │              │ ╰────┴────────╯    │ │
│                       │ ╰──────────────┴────────────────────╯ │
│ command_bar_text      │ #C4C9C6                               │
│ exit_esc              │ true                                  │
│                       │ ╭───────┬──────────────────╮          │
│ status                │ │       │ ╭──────┬───────╮ │          │
│                       │ │ error │ │ bg   │ red   │ │          │
│                       │ │       │ │ fg   │ white │ │          │
│                       │ │       │ │ attr │       │ │          │
│                       │ │       │ ╰──────┴───────╯ │          │
│                       │ ╰───────┴──────────────────╯          │
╰───────────────────────┴───────────────────────────────────────╯
```
it looks like the configuration of `explore` is all about style and not moving inside the menu

for example, i could imagine, inside the structure above in the config, a `motion` section, with a bunch of motions to tweak to one's likings :star_struck:

**Comment by fdncred** [comment]
All of these keybindings do not exist in explore right now. So, adding them may be a little tricky. But a quick win could be making synonyms for up, down, left, right, pgup, pgdown. That's rather trival to do. I already added `:q!` as quit because it's second nature to me.

We also still want to move more away from vim keybindings as a whole in explore. There's team members who don't like the modal ui ala vim. So, we may also eventually move further away from a vim-like interface.

**Comment by amtoine** [comment]
> All of these keybindings do not exist in explore right now. So, adding them may be a little tricky.
>
i see
for instance, `h`, `j`, `k` and `l` are not used by default in `explore`, they only get active when entering things likes command mode with `:` or the search with `/`.
i feel they migth be easily mapped to anything, right? 

> But a quick win could be making synonyms for up, down, left, right, pgup, pgdown. That's rather trival to do. I already added `:q!` as quit because it's second nature to me.
>
that's cool :yum: 

> We also still want to move more away from vim keybindings as a whole in explore. There's team members who don't like the modal ui ala vim. So, we may also eventually move further away from a vim-like interface.
>
yup why not! :ok_hand: 

in the end this issue is really about potential "configurable motion keybindings", not at all about the modal nature of `vim` and i wanted to make it extra clear that i dit NOT want to impose my view about making the `vim` motions a default in `explore`
> **Note**  
> that would be the ideal for me only and maybe the other people using `vim`

> **Warning**  
> BUT the meat of this feature request really is the new potential "motion" part of the `explore` config :relieved:

**Comment by 0xVeles** [comment]
I'd like to second configurable motion binds (& vim motion by default). Two of the programs I'm most likely to pipe something into for viewing on the CLI are less and bat, both of which have this behavior. I'd imagine that that's a common circumstance with a lot of people and not breaking that expectation would be great. Likewise supporting `q` as quit without the leader/command mode/whatever falls in the same category. All of which is addressed by configurable keybindings overall.

**Comment by dead10ck** [comment]
Yeah I think regardless of one's affinity for modal / vim key bindings, simply having configuration for the key bindings would offer a solution to everyone.

**Comment by texastoland** [comment]
I'd like to tackle this next week. It looks like copy pasta to implement. The code is really elegant. Literally everything else is already configurable.
